### PR TITLE
flag to set ServiceAccount token expiry

### PR DESCRIPTION
**Pull request by MatousJobanek**
https://redhat-internal.slack.com/archives/C06MJ2DVBU4/p1713353899754199

[KUBESAW-73](https://issues.redhat.com/browse/KUBESAW-73)

### CODE REVIEW

**Review by mfrancisc — APPROVED** [review]
Nice! 

Super fast ⚡

**Review by fbm3307 — APPROVED** [review]
LGTM

**Review by filariow — COMMENTED** [review]
nice, just some questions and nits

File: pkg/cmd/generate/cli_configs.go [hunk 1]
```diff
@@ -51,6 +52,7 @@ func NewCliConfigsCmd() *cobra.Command {
 
 	configDirPath := fmt.Sprintf("%s/src/github.com/kubesaw/ksctl/out/config", os.Getenv("GOPATH"))
 	command.Flags().StringVarP(&f.outDir, "out-dir", "o", configDirPath, "Directory where generated ksctl.yaml files should be stored")
+	command.Flags().IntVarP(&f.tokenExpirationDays, "token-expiration-days", "e", 365, "Expiration time of the ServiceAccount tokens in number of days")
```
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply pkg/cmd/generate/cli_configs.go]: Yeah, good point. TBH, I didn't think that anyone would even try to set some negative number :smile:

File: pkg/cmd/generate/cli_configs.go [hunk 2]
```diff
@@ -159,9 +162,10 @@ func writeKsctlConfigs(term ioutils.Terminal, configDirPath string, ksctlConfigs
 
 type generateContext struct {
 	ioutils.Terminal
-	newRESTClient   NewRESTClientFromConfigFunc
-	kubeSawAdmins   *assets.KubeSawAdmins
-	kubeconfigPaths []string
+	newRESTClient       NewRESTClientFromConfigFunc
+	kubeSawAdmins       *assets.KubeSawAdmins
+	kubeconfigPaths     []string
+	tokenExpirationDays int
```
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply pkg/cmd/generate/cli_configs.go:168]: In this case, I would just keep it as it is. It's the same as you would try to request a SA token directly from the cluster via `TokenRequest` API by setting `spec.expirationSeconds: 0`.

File: pkg/cmd/generate/cli_configs.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply pkg/cmd/generate/cli_configs.go]: done in 4a60955

**Pull request merged**